server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSUsageTelemetry.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.stats;
+
+import org.HdrHistogram.DoubleHistogram;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Service holding accumulated CCS search usage statistics. Individual cross-cluster searches will pass
+ * CCSUsage data here to have it collated and aggregated. Snapshots of the current CCS Telemetry Usage
+ * can be obtained by getting CCSTelemetrySnapshot objects (TODO: create that class (and determine the best name)).
+ */
+public class CCSUsageTelemetry {
```

Comment:
A bit confusing moment - this class is documented as "service" but the name doesn't have "Service" in it while UsageService does. Also SearchUsageHolder has "holder" in its name while CCSUsageTelemetry doesn't, despite them being quite analogous. Should we sync it?